test/lib/smartSurveyAPICall.test.js

```diff
@@ -0,0 +1,128 @@
+         that.createClient.should.have.been.calledWithExactly("a", "b")
+      })
+      it('calls this.client.getResponses with the correct params', function() {
+        that.client.getResponses.should.have.been.calledWithExactly("1",{ page: 1, pageSize: 25, includeLabels: false}, that.callback)
+      })
+      it('sets this.client to a new instance of SmartSurveyClient', function() {
+          let that = {};
+          // you can mock smartSurveyClientAPI as a constructor if you want to, but not essential
+          smartSurveyClientAPI.createClient.call(that, "a", "b");
+          that.client.should.be.instanceof(stubSmartSurvey);
+      })
+      it('calls the SmartSurveyClient constructor', function() {
+          var that = {};
+          // reassign the stubSmartSurvey with a stub to listen
+          stubSmartSurvey = sinon.stub();
+
+          // reproxyrequire because the stubSmartSurvey now has a stub
+          smartSurveyClientAPI = proxyquire('../../lib/smartSurveyAPICall', {
+            'smartsurvey-client': stubSmartSurvey
+          });
+
+          smartSurveyClientAPI.createClient.call(that, "1", "2");
+          stubSmartSurvey.should.have.been.called;
+      })
+  });
+
+      it('should call the callback with an error response when we receive an error from the api', function() {
+
+        let context = {
+          createClient: sinon.stub(),
+          client: {
+            getResponses: sinon.stub()
+          }
+        }
+        const callback = sinon.stub()
+
+        // Causes the stub to call the argument at the index as a callback function.
+        // stub.callsArg(2); stub calls the 3rd argument as a callback.
+        // callsArgWith like callsArg, but with arguments to pass to the callback.
+        context.client.getResponses.callsArgWith(2, 'error message', 1);
+        smartSurveyClientAPI.getData.call(context, "a", "b", undefined, callback)
+        callback.should.have.been.calledWith('error message', 1);
+      })
+
+      it('should call the callback without an error when we receive success from the api', function() {
+
+        let context = {
+          createClient: sinon.stub(),
+          client: {
+            getResponses: sinon.stub()
+          }
+        }
+        const callback = sinon.stub()
+
+        context.client.getResponses.callsArgWith(2, null, "good");
+        smartSurveyClientAPI.getData.call(context, "a", "b", undefined, callback)
+        callback.should.have.been.calledWith(null, "good");
+      })
+    describe('parseResponse', function() {
+      it('should return an error message if there is an error', function() {
+        parseResponse('error', null).should.equal('error')
+      })
+
+      it('should return the result if there is no error', function() {
+        const result = parseResponse(null, 'good response');
+        const expect = JSON.stringify('good response', null, 2);
+        result.should.deep.equal(expect);
+      })
```

Comment:
I'm fine with not having semicolons as long as:

- you're consistent
- you know how ASI works and all its edge cases